filibuster/instrumentation/redis/__init__.py

```diff
@@ -382,9 +385,62 @@ def _instrumented_redis_call(
             else:
                 debug("Instrumentation suppressed, skipping Filibuster instrumentation.")
 
-            response = func(*args, **kwargs)
-
-            return response
+            try:
+                ## TODO: don't need to do this
+                # debug("Setting Filibuster instrumentation key...")
+                # token = context.attach(context.set_value(_FILIBUSTER_SUPPRESS_REQUESTS_INSTRUMENTATION_KEY, True))
+
+                if not should_inject_fault:
+                    # no need to propagate vclock and origin vclock forward,
+                    # since redis-server won't use it
+                    result = func(*args, **kwargs)
+                elif should_inject_fault and not should_abort:
+                    # If we should delay the request to simulate timeouts, do it.
+                    if should_sleep_interval != 0:
+                        time.sleep(should_sleep_interval)
+
+                    # no need to propagate vclock and origin vclock forward,
+                    # since redis-server won't use it
+                    result = func(*args, **kwargs)
+                else:
+                    # Return entirely fake response and do not make request.
+                    #
+                    # Since this isn't a real result object, there's some attribute that's
+                    # being set to None and that's causing -- for these requests -- the opentelemetry
+                    # to not be able to report this correctly with the following error in the output:
+                    #
+                    # "Invalid type NoneType for attribute value.
+                    # Expected one of ['bool', 'str', 'int', 'float'] or a sequence of those types"
+                    #
+                    # I'm going to ignore this for now, because if we reorder the instrumentation
+                    # so that the opentelemetry is installed *before* the Filibuster instrumentation
+                    # we should be able to avoid this -- it's because we're returning an invalid
+                    # object through the opentelemetry instrumentation.
+                    #
+                    result = Response()
+            except Exception as exc:
+                exception = exc
+                result = getattr(exc, "response", None)
+            finally:
+                debug("Removing instrumentation key for Filibuster.")
+                # context.detach(token)
+
+            # Result was an actual response.
+            if isinstance(result, Response) and (exception is None or exception == "None"):
```

Comment:
Again, I think this is instance call needs to be changed to reflect the proper type.